server/assetstore.py

```diff
@@ -34,6 +34,43 @@
 DB_INFO_KEY = 'databaseMetadata'
 
 
+class DatabaseAssetstoreFile(dict):
+    """
+    This wraps a Girder file object dictionary so that the size parameter can
+    be determined only when actually necessary.
+    """
+    def __init__(self, file, adapter, *args, **kwargs):
+        """
+        Create a file dictionary where 'size' and 'mimeType' are determined
+        lazily.
+
+        :param file: the original file dictionary.
+        :param adapter: the owning assetstore adapter.
+        """
+        self._file = file
```

Comment:
I was following the convention in Girder's `girder/models/file.py` of using `file` as a parameter.